crates/grpc-util/src/lib.rs

```diff
@@ -39,3 +48,119 @@ pub fn create_grpc_channel_from_advertised_address(
     };
     Ok(channel)
 }
+
+#[derive(Debug, thiserror::Error)]
+pub enum Error {
+    #[error("failed binding to address '{address}': {source}")]
+    TcpBinding {
+        address: SocketAddr,
+        #[source]
+        source: hyper::Error,
+    },
+    #[error("failed opening uds '{uds_path}': {source}")]
+    UdsBinding {
+        uds_path: String,
+        #[source]
+        source: io::Error,
+    },
+    #[error("failed running grpc server: {0}")]
+    Running(#[from] hyper::Error),
+}
+
+pub async fn run_hyper_server<S, B, F>(
+    bind_address: BindAddress,
+    service: S,
+    shutdown_signal: F,
+) -> Result<(), Error>
+where
+    S: hyper::service::Service<http::Request<hyper::Body>, Response = hyper::Response<B>>
+        + Send
+        + Clone
+        + 'static,
+    S::Error: Into<Box<dyn std::error::Error + Send + Sync>>,
+    S::Future: Send,
+    B: HttpBody + Send + 'static,
+    B::Data: Send,
+    B::Error: Into<Box<dyn std::error::Error + Send + Sync>>,
+    F: Future<Output = ()>,
+{
+    match bind_address {
+        BindAddress::Uds(uds_path) => {
+            let unix_listener = UnixListener::bind(&uds_path).map_err(|err| Error::UdsBinding {
+                uds_path: uds_path.clone(),
+                source: err,
+            })?;
+            let acceptor =
+                hyper::server::accept::from_stream(UnixListenerStream::new(unix_listener));
+
+            info!(
+                    uds.path = %uds_path,
+                    "Node server listening");
+
+            run_server(acceptor, service, shutdown_signal).await?
+        }
+        BindAddress::Socket(socket_addr) => {
+            run_tcp_server(socket_addr, service, shutdown_signal).await?
+        }
+    }
+
+    Ok(())
+}
+
+async fn run_tcp_server<S, B, F>(
+    socket_addr: SocketAddr,
+    service: S,
+    shutdown_signal: F,
+) -> Result<(), Error>
+where
+    S: hyper::service::Service<http::Request<hyper::Body>, Response = hyper::Response<B>>
+        + Send
+        + Clone
+        + 'static,
+    S::Error: Into<Box<dyn std::error::Error + Send + Sync>>,
+    S::Future: Send,
+    B: HttpBody + Send + 'static,
+    B::Data: Send,
+    B::Error: Into<Box<dyn std::error::Error + Send + Sync>>,
+    F: Future<Output = ()>,
+{
+    let acceptor = AddrIncoming::bind(&socket_addr).map_err(|err| Error::TcpBinding {
+        address: socket_addr,
+        source: err,
+    })?;
+
+    info!(
+        net.host.addr = %acceptor.local_addr().ip(),
+        net.host.port = %acceptor.local_addr().port(),
+        "Metadata store server listening"
+    );
```

Comment:
Similar comment to the one above. This could be an artifact of an earlier version of this code maybe?

Reply:
Good point. Will fix it.